static/js/publisher/form/banner.js

```diff
@@ -0,0 +1,219 @@
+import React, { Fragment } from "react";
+import PropTypes from "prop-types";
+
+import FileInput from "../form/fileInput";
+import AccordionHelp from "./AccordionHelp";
+
+class Banner extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.bannerImageChangeHandler = this.bannerImageChangeHandler.bind(this);
+    this.handleRemoveImageClick = this.handleRemoveImageClick.bind(this);
+
+    this.keyboardEvent = this.keyboardEvent.bind(this);
+    this.toggleFocus = this.toggleFocus.bind(this);
+
+    this.state = {
+      bannerImage: this.props.bannerImage,
+      focused: false,
+      errors: {}
+    };
+  }
+
+  bannerImageChangeHandler(files) {
+    const { updateImageState } = this.props;
+    const file = files[0];
+
+    if (file.errors) {
+      this.setState({
+        errors: {
+          [file.name]: file.errors
+        }
+      });
+    } else {
+      const url = URL.createObjectURL(file);
+
+      this.setState({
+        errors: {},
+        bannerImage: {
+          url: url
+        }
+      });
+
+      updateImageState({
+        url: url,
+        file,
+        name: file.name,
+        status: "new",
+        type: "banner"
+      });
+    }
+  }
+
+  handleRemoveImageClick(e) {
```

Comment:
Not sure if there is a simple way to abstract it, but seems like we are duplicating a lot of same logic in icon/screenshots/banner components.
While they all reuse file input component, the 'wrapper' still reimplements a lot of similar stuff.

Reply:
Yeah I was thinking of doing a follow up to do abstract to another component (`MediaEdit` or something) - along with the scss that's related.